FHEM/14_SD_WS.pm

```diff
@@ -1525,6 +1527,52 @@ sub SD_WS_Parse {
                               return 0;
                             }
 
+                            return 1;
+                          }, 
+    },
+    126 => {
+        # rain gauge Fine Offset WH40, aka Ambient Weather, aka ecowitt
+        # ------------------------------------------------------------------------------------------
+        #          Byte: 00 01 02 03 04 05 06 07 08 09 10 11 12 13 
+        #        Nibble: 01 23 45 67 89 01 23 45 67 89 01 23 45 67 
+        # aa aa aa 2d d4 40 01 1C DF 8F 00 00 97 62 20 A6 80 28 01       -> ID  0x11cdf
+        #                40 01 3E 3C 90 00 00 10 5B A0 2A                -> ID: 0x13e3c
+        #                FF II II II ?? RR RR XX AA ?? ?? ?? ?? ?? ?? ??
+        # FF:   Family code always 0x40
+        # II:   ID (1 byte)
+        # RR:   the rain bucket tip count => 0.1mm increments
+        # XX:   CRC8 of the preceding 5 bytes (Polynomial 0x31, Initial value 0x00, Input not reflected, Result not reflected)
+        # SS:   Sum-8 of the preceding 5 bytes 
+        # ??:   Unknown Data
+
+        sensortype => 'WH40',
+        model      => 'SD_WS_126_R',
+        prematch   => sub {my ($rawData,undef) = @_; return 1 if ($rawData =~ /^40/); },
+        id         => sub {my ($rawData,undef) = @_; return (substr($rawData,2,6));},
+        rain       => sub {my ($rawData,undef) = @_; return 0.1 * substr($rawData,10,4); },
```

Comment:
rain ist sicher nicht BCD-kodiert. Ich sehe im Quellcode auch ein 0x000a bei den Beispielen.

Reply:
Das weiss ich aktuell noch nicht.
Möglich, dass es in einer dezimal Zahl umgerechnet werden muss.

Reply:
Sicher:

```
2023.04.03 14:11:11 4: sduino_dummy: get rawmsg: MN;D=4000CD6F10000ABF550002DB0000;
2023.04.03 14:11:11 4: sduino_dummy: Parse_MN, Found 2-FSK Protocol id 126 -> WH40 with match (?^:^40)
2023.04.03 14:11:11 4: sduino_dummy: SD_WS_Parse protocol 126, rawData 4000CD6F10000ABF550002DB0000
2023.04.03 14:11:11 1: PERL WARNING: Argument "000A" isn't numeric in multiplication (*) at ./FHEM/14_SD_WS.pm line 1577.
2023.04.03 14:11:11 4: sduino_dummy: SD_WS_Parse decoded protocol-id 126 (WH40), sensor-id 00CD6F
```

Reply:
habe es angepasst